cmd/cluster-agent/api/server.go

```diff
@@ -129,5 +129,6 @@ func isExternalPath(path string) bool {
 		strings.HasPrefix(path, "/api/v1/tags/pod/") && (len(strings.Split(path, "/")) == 6 || len(strings.Split(path, "/")) == 8) ||
 		strings.HasPrefix(path, "/api/v1/tags/node/") && len(strings.Split(path, "/")) == 6 ||
 		strings.HasPrefix(path, "/api/v1/clusterchecks/") && len(strings.Split(path, "/")) == 6 ||
-		strings.HasPrefix(path, "/api/v1/endpointschecks/") && len(strings.Split(path, "/")) == 6
+		strings.HasPrefix(path, "/api/v1/endpointschecks/") && len(strings.Split(path, "/")) == 6 ||
+		path == "/api/v1/tags/cf/apps"
```

Comment:
shouldn't node agents query this by declaring their node names and only get tags that correspond to the apps running on that specific node? we can avoid sending huge payloads and simplify the tag-filtering logic in the node agents

Reply:
Good point, I'll add that in

Reply:
thanks, please make sure you validate the url by checking the prefix and the value of `len(strings.Split(path, "/"))`